numba/unicode.py

```diff
@@ -622,6 +623,34 @@ def join_str_impl(sep, parts):
         return join_str_impl
 
 
+@overload_method(types.UnicodeType, 'zfill')
+def unicode_zfill(string, width):
+    if not isinstance(width, types.Integer):
+        raise TypingError("<width> must be an Integer")
+
+    def zfill_impl(string, width):
+
+        str_len = len(string)
+
+        if width <= str_len:
+            return string
+
+        if str_len != 0:
+            first_char = string[0]
```

Comment:
This is still a bit weird. If this were pure python, and the length would be 0, the code would fail later on complaining that `first_char` is not defined. However, since this is being compiled, first_char is initialized to whatever the default for it's type and thus the if statement later doesn't crash. I would propose to fix this. There are a few ways you can do this but short readable code is preferred.

Reply:
@esc DO you think following could be good enough?
```
        first_char = ''
        if str_len != 0:
            first_char = string[0]
```

Reply:
How about this:

```
first_char = string[0] if str_len else ''
```